GoogleForCreators/web-stories-wp
# Design Panel: Add contrast warning

## Context


We want to notify users when they choose a text color and background color combination that doesn't meet accessibility. 

## Summary


When a user chooses a font color or background color that doesn't have sufficient contrast, we want to show some warning text where they pick the colors. Now inside the Design Panel -> Text we are showing a warning message. We also are making use of our `useLiveRegion` hook to announce the message as it appears. 
I also added the warning to the `background` panel. 

## Relevant Technical Choices


Created a `Warning` component that can be reused with custom messages as we find other areas where we want to show feedback sooner than in the checklist. 
Added a constant for the warning messages under `ACCESSIBILITY_COPY`
Moved `getFailingPages` into the `checklist/checks/pageBackgroundLowTextContrast/check.js` file so that we can use it in both in the checklist check, and here in the panel. Also, renamed it to `getPagesWithFailedContrast`
\`pageBackgroundTextLowContrast(page)` (which is called inside `getPagesWithFailedContrast` ) now returns the offending elements' id. This way we know exactly which elements should be flagged.

The reason I am checking the whole page for contrast issues is because we could have more than one text selection. If either of them doesn't have enough contrast, we should show the message. Plus, we already need to grab all elements from the page to determine which ones are considered as part of the background. 

## To-do
Some background images may flag for low contrast. I am not sure how we want to accommodate for this. 


## User-facing changes
<img width="1122" alt="Screen Shot 2021-12-31 at 12 50 11 PM" src="https://user-images.githubusercontent.com/1820266/147837565-293663f4-290a-4d1c-a398-a9765ac11abf.png">
<img width="1121" alt="Screen Shot 2021-12-31 at 12 50 22 PM" src="https://user-images.githubusercontent.com/1820266/147837566-3a853aa3-d20b-4e02-b4e7-1a20e46d643b.png">
<img width="1084" alt="Screen Shot 2021-12-31 at 12 50 42 PM" src="https://user-images.githubusercontent.com/1820266/147837567-9192ea6a-b473-4765-96b5-239c3cfa9c35.png">
<img width="1029" alt="Screen Shot 2021-12-31 at 12 53 21 PM" src="https://user-images.githubusercontent.com/1820266/147837582-4eb88d0b-4ef8-44f6-b0ce-294baad6c139.png">



## Testing Instructions






This PR can be tested by following these steps:

1. In editor add some text. 
2. Change the font color to match the background. 
3. See the warning appear. 
4. Change it back and the warning should disappear. 
5. Check that the new message is announced. 
6. Turn on voice over, or other reader. 
7. Follow above steps, you should hear the message announced. 
8. In addition, the warning should appear on the background panel. 
9. Click the background on the canvas. The background panel should open. 
10. Change the color to low contrast and you should see the warning message. 
11. The check should still work as normal in the Pre-publish Checklist as well. 


## Reviews

### Does this PR have a security-related impact?
no


### Does this PR change what data or activity we track or use?
no


### Does this PR have a legal-related impact?
no


## Checklist




---



Fixes #7230
